src/daemon/engine/svc/node/exec_cmd_service.cpp

```diff
@@ -0,0 +1,265 @@
+//
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+//
+
+#include "exec_cmd_service.hpp"
+
+#include "engine_helpers.hpp"
+#include "ipc/channel.hpp"
+#include "ipc/server_router.hpp"
+#include "logging.hpp"
+#include "svc/node/exec_cmd_spec.hpp"
+#include "svc/svc_helpers.hpp"
+
+#include <uavcan/node/ExecuteCommand_1_3.hpp>
+
+#include <cetl/pf17/cetlpf.hpp>
+#include <libcyphal/presentation/client.hpp>
+#include <libcyphal/presentation/presentation.hpp>
+#include <libcyphal/presentation/response_promise.hpp>
+
+#include <cerrno>
+#include <cstdint>
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+
+namespace ocvsmd
+{
+namespace daemon
+{
+namespace engine
+{
+namespace svc
+{
+namespace node
+{
+namespace
+{
+
+class ExecCmdServiceImpl final
+{
+public:
+    using Spec    = common::svc::node::ExecCmdSpec;
+    using Channel = common::ipc::Channel<Spec::Request, Spec::Response>;
+
+    explicit ExecCmdServiceImpl(const ScvContext& context)
+        : context_{context}
+    {
+    }
+
+    void operator()(Channel&& channel, const Spec::Request& request)
```

Comment:
This is mysterious. Perhaps some docs to explain why this is a function call operator and not a named function?

Reply:
ok